lib/algos/list/delimiterMaster.js

```diff
@@ -165,14 +170,54 @@ class DelimiterMaster extends Delimiter {
         return super.filter(obj);
     }
 
+    allowDelimiterRangeSkip(key) {
+        if (!this.prefixKeySeen) {
+            return true;
+        }
+        // prefix = prefix, key = prefix/. Can skip since key will be part of commonPrefixes.
+        if (key[key.length - 1] === this.delimiter && key.length > this.prefix.length && key.startsWith(this.prefix)) {
+            return true;
+        }
+        const lastIdx = key.lastIndexOf(this.delimiter);
+        return lastIdx === - 1 || !(key.slice(0, lastIdx + 1) === this.prefix);
+    }
+
+    nextSkipRange(key) {
+        // const baseIndex = this.prefix ? this.prefix.length : 0;
+        // const delimiterIndex = key.indexOf(this.delimiter, baseIndex);
+        // if (delimiterIndex >= 0) {
+        //     const commonPrefix = getCommonPrefix(key, this.delimiter, baseIndex);
+        //     console.log('inside can add common', {key, commonPrefix, prefix: this.prefix});
+        //     if (this.CommonPrefixes.indexOf(commonPrefix) === -1
+        //             && this[this.nextContinueMarker] !== commonPrefix) {
+        //         return true;
+        //     }
+        // }
+        // return false;
+
+        const delimEnd = this.prefix && this.prefix.endsWith(this.delimiter); // prefix/
+        if (!delimEnd) {
+            return null;
+        }
+        const delimIndex = this.prefix.lastIndexOf(this.delimiter);
+        const keySuffix = key.slice(delimIndex + 1);
+
+
+        if (keySuffix.endsWith(this.delimiter)) { // prefix/foo/
+            return key;
+        }
+        return key.slice(0, key.lastIndexOf(this.delimiter) + 1); // prefix/foo/1
```

Comment:
Test should not be needed (unless there is a real efficiency reason to do the first check but I don't think so)

```suggestion
        return key.slice(0, key.lastIndexOf(this.delimiter) + 1); // prefix/foo/bar -> prefix/foo/
```